2009-02-13  Michael Tautschnig  <mt@debian.org>

	* lib/get-config-dir-svn: Added support for password in SVN URL (thanks Robin
		Powell) (closes: #515057)
Index: trunk/lib/get-config-dir-svn
===================================================================
--- trunk.orig/lib/get-config-dir-svn
+++ trunk/lib/get-config-dir-svn	
@@ -11,9 +11,13 @@
 
 # matched string: "svn://user@host/svnpath"
 protocol=$(expr match "$FAI_CONFIG_SRC" '\([^:]*\)://')
-username=$(expr match "$FAI_CONFIG_SRC" '[^:]*://\([^@]*\)@')
+# username may or may not be followed by a password
+username=$(expr match "$FAI_CONFIG_SRC" '[^:]*://\([^@:]*\)[^@]*@')
 if [ -n "$username" ] ; then
   svnpath=$(expr match "$FAI_CONFIG_SRC" '[^:]*://[^@]\+@\([^[:space:]]\+\)')
+  # password definitely is preceded by a username
+  password=$(expr match "$FAI_CONFIG_SRC" '[^:]*://[^@:]*:\([^@]*\)@')
+  [ -n "$password" ] && pass="--password $password"
 else
   svnpath=$(expr match "$FAI_CONFIG_SRC" '[^:]*://\([^[:space:]]\+\)')
 fi
@@ -49,10 +53,10 @@
 if [ -d "$FAI/.svn" ] ; then
    echo "Updating SVN in $FAI"
    cd $FAI
-   svn up $user | grep -v 'Updated to revision' > $LOGDIR/getconf.log
+   svn up $user $pass | grep -v 'Updated to revision' > $LOGDIR/getconf.log
    task_error 701 $?
 else 
    echo "Checking out SVN"
-   svn co $user $svnurl $FAI | grep -v 'Checked out revision' > $LOGDIR/getconf.log
+   svn co $user $pass $svnurl $FAI | grep -v 'Checked out revision' > $LOGDIR/getconf.log
    task_error 702 $?
 fi
